### PR TITLE
Updated the LimitWithoutOrderByChecker to handle additional node types

### PR DESCRIPTION
**Pull request by anujjamwal**
Test plan - (Please fill in how you tested your changes)
Added Unit Tests for the LimitWithoutOrderByChecker class

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java [hunk 1]
```diff
@@ -395,62 +392,60 @@ protected Boolean visitQuerySpecification(QuerySpecification node, Integer level
...
         {
-            if (node.getWith().isPresent()) {
-                With with = node.getWith().get();
-                for (WithQuery query : with.getQueries()) {
-                    if (process(query.getQuery(), level + 1)) {
-                        return true;
-                    }
-                }
+            if (node.getChildren().stream().anyMatch(n -> process(n, level))) {
+                return true;
```
Comment (ajaygeorge) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java:396]: One thing to note is `getChildren` is not precomputed and could be a costly call. The implementation from Query.java is something like this
```
  public List<Node> getChildren()
    {
        ImmutableList.Builder<Node> nodes = ImmutableList.builder();
        with.ifPresent(nodes::add);
        nodes.add(queryBody);
        orderBy.ifPresent(nodes::add);
        offset.ifPresent(nodes::add);
        return nodes.build();
    }
 ```
 I think we are only interested in the with clause here and queryBody here ?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java [hunk 2]
```diff
@@ -466,32 +461,31 @@ protected Boolean visitTable(Table node, Integer level)
...
+            return false;
         }
 
         @Override
-        protected Boolean visitCreateView(CreateView node, Integer level)
+        protected Boolean visitOrderBy(OrderBy node, Integer context)
         {
-            return process(node.getQuery(), level);
+            return false;
         }
```
Comment (ajaygeorge) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java:485]: I can see these got taken out. Can you help me understand how this gets handled in the new code.

Comment (anujjamwal) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java:485]: The getChildren() call in `CreateTableAsSelect` and `CreateView` returns immutable list containing `Query`.
Both `visitCreateTableAsSelect` and `visitCreateView`  methods invoke  `visitStatement(node, context);` which in turn invokes `visitNode(node, context);`. visitNode implementation here processes each child so visitQuery gets invoked

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java (hunk 1 above)
Comment (anujjamwal) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java:396]: This is true for all the classes. All of the classes create new immutable lists on each invocation. We can refactor those to eagerly create children list but that would be a larger refactor + need to consider the memory impact on the critical query execution path.

**Comment by ajaygeorge** [comment]
Adding @kaikalur to take a look.

**Review by kaikalur — APPROVED** [review]
Stylistically we don't use final for local variables. So just remove them all.

**Comment by ajaygeorge** [comment]
@anujjamwal can you please squash the commits into one . Also remove the Release notes and just keep the No Release notes section.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java [hunk 3]
```diff
@@ -376,17 +375,15 @@ private void postEvent(VerifierQueryEvent event)
...
     }
 
-    private class LimitWithoutOrderByChecker
+    @VisibleForTesting
```
Comment (highker) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java:378]: no need to have this annotation

File: presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestVerificationManager.java [hunk 4]
```diff
@@ -224,4 +227,251 @@ private VerificationManager getVerificationManager(List<SourceQuery> sourceQueri
...
+                    "  FROM " +
+                    "    source_table TABLESAMPLE BERNOULLI(0.1) " +
+                    "  WHERE " +
+                    "    ds = '2021-11-20' " +
+                    "    AND pair_id IS NOT NULL " +
+                    "  GROUP BY 1, 2, 3 " +
+                    ")";
+            final Statement statement = SQL_PARSER.createStatement(query, PARSING_OPTIONS);
+            final VerificationManager.LimitWithoutOrderByChecker checker = new VerificationManager.LimitWithoutOrderByChecker();
+            final Boolean result = checker.process(statement, 0);
```
Comment (highker) [inline presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestVerificationManager.java:250]: remove all the `final` keywords in the tests. We don't usually use final in the body of methods

File: presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestVerificationManager.java [hunk 5]
```diff
@@ -224,4 +227,251 @@ private VerificationManager getVerificationManager(List<SourceQuery> sourceQueri
...
+
+            assertFalse(result);
```
Comment (highker) [inline presto-verifier/src/test/java/com/facebook/presto/verifier/framework/TestVerificationManager.java:252]: inline these two lines. Same for all other functions

**Review by highker — APPROVED** [review]
As @ajaygeorge mentioned, please squash all commits into one

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java (hunk 3 above)
Comment (anujjamwal) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/VerificationManager.java]: Removed the annotation.

**Review by tdcmeehan — APPROVED** [review]

**Review by ajaygeorge — APPROVED** [review]

**Comment by highker** [comment]
@anujjamwal, could you help to amend and force push again? The CLA somehow was not triggered.

**Pull request merged**